src/mono/wasi/runtime/main.c

```diff
@@ -21,8 +21,38 @@ int main(int argc, char * argv[]) {
 #endif
 	mono_wasm_load_runtime("", 0);
 
+	int arg_ofs = 0;
+#ifdef WASM_SINGLE_FILE
```

Comment:
this is the same problem we have with shell on emscripten. 
I would prefer to separate the "runtime options" from "application arguments" by ` -- `. 
In this case there are no "runtime options" because we know which assembly to load. 

it would be `wasmtime run --dir . -- --run MainAssembly.wasm -- [args]`

In the non-single file the syntax would be 
it would be `wasmtime run --dir . -- -- [args]`

Note, that wasmtime also has ` -- ` it's own, to solve the same problem.

This would allow us in the future to add other runtime options, without counting positions of the arguments.
For example single-file
`wasmtime run --dir . -- --soft-breakpoints -- [args]`

where `--soft-breakpoints` is example of Mono option

Reply:
Cleanly delineating arguments sounds like a good idea. I can explore that in a follow up PR. This PR fixes a regression and gets it passing on CI.